src/classes/STG_PanelRDHealthCheck.cls

```diff
@@ -0,0 +1,480 @@
+            RD_Constants.INSTALLMENT_PERIOD_YEARLY,
+            RD_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH
+        };
+        public Set<String> customInstallmentPeriods = new Set<String>();
+
+        public Set<String> status = new Set<String>{
+            RD_Constants.OPEN_ENDED_STATUS_OPEN,
+            RD_Constants.OPEN_ENDED_STATUS_CLOSED,
+            RD_Constants.OPEN_ENDED_STATUS_NONE
+        };
+
+        public Set<String> scheduleTypes = new Set<String>{
+            RD_Constants.SCHEDULE_TYPE_MULTIPLY_BY,
+            RD_Constants.SCHEDULE_TYPE_DIVIDE_BY
+        };
+
+        public SupportedLegacyRDValues() {
+            populateCustomInstallmentPeriods();
+        }
+
+        private void populateCustomInstallmentPeriods() {
+            for (npe03__Custom_Installment_Settings__c ci : STG_Panel.stgService.listRDCustomInstallment) {
+                installmentPeriods.add(ci.Name);
+                customInstallmentPeriods.add(ci.Name);
+            }
+        }
+    }
+
+    /**
+     * @description Contains field values supported for the enhanced Recurring Donations
+     */
+    public class SupportedEnhancedRDValues {
+        public Set<String> installmentPeriods = new Set<String>{
+            RD2_Constants.INSTALLMENT_PERIOD_DAILY,
+            RD2_Constants.INSTALLMENT_PERIOD_WEEKLY,
+            RD2_Constants.INSTALLMENT_PERIOD_MONTHLY,
+            RD2_Constants.INSTALLMENT_PERIOD_YEARLY,
+            RD2_Constants.INSTALLMENT_PERIOD_FIRST_AND_FIFTEENTH
+        };
+
+        public Set<String> status = new Set<String>{
+            RD2_Constants.STATUS_ACTIVE,
+            RD2_Constants.STATUS_CLOSED,
+            RD2_Constants.STATUS_LAPSED
+        };
+
+        public Set<String> recurringTypes = new Set<String>{
+            RD2_Constants.RECURRING_TYPE_OPEN
+        };
+
+        public Set<String> dayOfMonthValues {
+            get {
+                if (dayOfMonthValues == null) {
+                    dayOfMonthValues = getSupportedDayOfMonth();
+                }
+                return dayOfMonthValues;
+            }
+            private set;
+        }
+
+        private Set<String> getSupportedDayOfMonth() {
+            Set<String> values = new Set<String>();
+            for (Integer i = 0; i < 29; i++) {
+                values.add(String.valueOf(i));
+            }
+
+            values.add(RD2_Constants.DAY_OF_MONTH_LAST_DAY);
+            return values;
+        }
+    }
+
+    /**
+     * @description Builds and collects health check results
+     */
+    public class HealthCheckResult {
+
+        private final String statusError = STG_PanelHealthCheck_CTRL.statusError;
+        private final String statusSuccess = STG_PanelHealthCheck_CTRL.statusSuccess;
+
+        private STG_PanelHealthCheck_CTRL ctrl;
+        private String scope;
+
+        public HealthCheckResult(STG_PanelHealthCheck_CTRL ctrl) {
+            this.ctrl = ctrl;
+        }
+
```

Comment:
Missing ApexDoc on the new inner class methods.

Reply:
Thanks. Done.